sonic-net/sonic-gnmi
# Streaming events URL support "not to use cache"

#### Why I did it
Only one gNMI client can use cache at any time. 
The external clients that stream events reliably need to use cache.
While an external client is connected, if we need to use some test client (e.g. gnmi_cli), it could be used provided it does not use cache.

#### How I did it
Added support for a new URL parameter [usecache=false].  This parameter will *disable* use of cache.

#### How to verify it
1) Have no gNMI client
2) Ensure some events are published (may use events_tool to manually publish)
3) Run gnmi_cli with [usecache=false]
4) Note, it does not receive/spew any events from past.
5) Publish some events (may use events_tool)
6) See the client receive these
7) Now run a client with cache use *not* disabled
8) Note, it receives all events including those from step 2

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
